src/tuning/tuning.hpp

```diff
@@ -93,13 +98,46 @@ void Tuner(int argc, char* argv[]) {
 
   // Use full-search to explore all parameter combinations or random-search to search only a part of
   // the parameter values. The fraction is set as a command-line argument.
-  if (args.fraction == 1.0 || args.fraction == 0.0) {
+  #ifdef XGEMM_EXEC
```

Comment:
Can you try to make this a bit nicer? First of all no #ifdef's please! Second, try to integrate both cases in a nice way. Now there's 3 places where `UseFullSearch` is set! You can also just introduce one new command-line argument which is "use_pso" and is set to true per default and then have the parameters coded into the code? Not sure how much we would want to change them, but you probably know better than I do.